### PR TITLE
Fix network.transport and network.protocol issues in m365_defender

### PR DESCRIPTION
**Pull request by peterydzynski**
## Proposed commit message
The `network.transport` and `network.protocol` fields are not being set properly. Often the transport value (udp/tcp) is getting set to the protocol field and the protocol is never set at all. The changes I made to the `test-device.log-expected.json` file should demonstrate these issues. 

I also added handling for missing protocols and added example logs for the ones I had the ability to produce. 

As a side note, I am happy to convert the entire logic for handling transport/protocol to a single script processor but I figured using the built in processors is easier for maintainability.  

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

**Review by efd6 — COMMENTED** [review]
Please add a changelog entry and bump the patch version in manifest.yml.

**Comment by peterydzynski** [comment]
Whoops forgot to do those. Should be updated @efd6!

File: packages/m365_defender/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.14.2"
+  changes:
+    - description: Fix network.transport and network.protocol.
```
Comment (efd6) [inline packages/m365_defender/changelog.yml:4]:
```suggestion
    - description: Fix `network.transport` and `network.protocol` processing.
```

**Review by efd6 — COMMENTED** [review]
Nit then LGTM

**Comment by peterydzynski** [comment]
Hey @efd6, is this good to go?

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**